src/Microsoft.AspNet.Mvc.TagHelpers/SelectTagHelper.cs

```diff
@@ -94,7 +95,8 @@ public override void Init(TagHelperContext context)
 
             // Whether or not (not being highly unlikely) we generate anything, could update contained <option/>
             // elements. Provide selected values for <option/> tag helpers.
-            context.Items[typeof(SelectTagHelper)] = _currentValues;
+            var currentValues = _currentValues == null ? null : new CurrentValues(_currentValues);
+            context.Items[typeof(SelectTagHelper)] = currentValues;
```

Comment:
Current `OptionTagHelper` doesn't need a `CurrentValues` instance when `Values` is empty. But allocating one in that case keeps the semantics consistent with `IHtmlGenerator.GetCurrentValues()` and the expression value.